build.go

```diff
@@ -34,46 +34,123 @@ import (
 )
 
 const (
+	// the lifecycle image that will be used for the analysis and export phases
+	// when using an untrusted builder
 	lifecycleImageRepo                   = "buildpacksio/lifecycle"
 	minLifecycleVersionSupportingCreator = "0.7.4"
 	prevLifecycleVersionSupportingImage  = "0.6.1"
 	minLifecycleVersionSupportingImage   = "0.7.5"
 )
 
+// A Lifecycle satisfies the Cloud Native Buildpacks Lifecycle specification.
+// Implementations of the Lifecycle must execute the following phases by calling the
+// phase-specific lifecycle binary in order:
+//
+//  Detection:         /cnb/lifecycle/detector
+//  Analysis:          /cnb/lifecycle/analyzer
+//  Cache Restoration: /cnb/lifecycle/restorer
+//  Build:             /cnb/lifecycle/builder
+//  Export:            /cnb/lifecycle/exporter
 type Lifecycle interface {
+	// Execute is responsible for invoking each of these binaries,
+	// with the desired configuration.
 	Execute(ctx context.Context, opts build.LifecycleOptions) error
 }
 
+// BuildOptions defines configuration settings for a Build.
 type BuildOptions struct {
-	Image              string // required
-	Builder            string // required
-	Registry           string
-	AppPath            string              // defaults to current working directory
-	RunImage           string              // defaults to the best mirror from the builder metadata or AdditionalMirrors
-	AdditionalMirrors  map[string][]string // only considered if RunImage is not provided
-	Env                map[string]string
-	Publish            bool
-	ClearCache         bool
-	TrustBuilder       bool
-	Buildpacks         []string
-	ProxyConfig        *ProxyConfig // defaults to  environment proxy vars
-	ContainerConfig    ContainerConfig
+	// required. Name of output image
+	Image string
+
+	// required. Builder image name.
+	Builder string
+
+	// URI to a buildpack registry. Used to
+	// add buildpacks to a build.
+	Registry string
+
+	// path to application bits, defaults to current working directory
+	AppPath string
+
+	// Specify the run image.
+	RunImage string
+
+	// If Run Image is empty, it is set to the the 'best' mirror using Builder metadata and AdditionalMirrors.
+	// where 'best' is defined as:
+	//  - if the builder or Additional mirrors has a registry that matches the registry in
+	//    Image, it is 'best'
+	//  - otherwise if there are no matches, use the builder metadata
```

Comment:
I believe this recently changed to have different behavior for `--publish` vs. daemon. It might be worth calling that out.